nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,913 @@
+            (attr_name, model.__dict__[attr_name])
+        for attr_name in selected_attributes
+        if attr_name in model.__dict__
+        )
+    mask_img = display_attributes.get('mask_img', None)
+    img_types = (nibabel.nifti1.Nifti1Image,
+                 nibabel.nifti2.Nifti2Image
+                 )
+    if mask_img and isinstance(mask_img, img_types):
+        mask_img = '{} with shape {}'.format(type(mask_img),
+                                                  mask_img.shape)
+    display_attributes['mask_img'] = mask_img
+    model_attributes = pd.DataFrame.from_dict(display_attributes,
+                                                    orient='index',
+                                                    )
+    attribute_names_with_units = {
+        attribute_name_: attribute_name_+' ({})'.format(attribute_unit_)
+        for attribute_name_, attribute_unit_
+        in attribute_units.items()
+        }
+    model_attributes.rename(index=attribute_names_with_units,
+                                  inplace=True)
+    return model_attributes
+
+
+def make_stat_maps(model, contrasts):
+    """ Given a model and contrasts, return the corresponding z-maps
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        Must have a fitted design matrix(ces).
+    
+    contrasts: Dict[str, ndarray or str]
+        Dict of contrasts for a first or second level model.
+        Corresponds to the contrast_def for the FirstLevelModel [1]_
+        & second_level_contrast for a SecondLevelModel [2]_ .
+
+    
+    Returns
+    -------
+    statistical_maps: Dict[str, niimg]
+        Dict of statistical z-maps keyed to contrast names/titles.
+        
+    See Also
+    --------
+    .. [1] nistats.first_level_model.FirstLevelModel.compute_contrast
+    .. [2] nistats.second_level_model.SecondLevelModel.compute_contrast
+    """
+    statistical_maps = {contrast_id: model.compute_contrast(contrast_val)
+                        for contrast_id, contrast_val in contrasts.items()
+                        }
+    return statistical_maps
+
+
+def _dmtx_to_svg_url(design_matrices):
+    """ Accepts a FirstLevelModel or SecondLevelModel object
+    with fitted design matrices & generates SVG Image URL,
+    which can be inserted into an HTML template.
+    
+    Parameters
+    ----------
+    design_matrices: List[pd.Dataframe]
+        Design matrices computed in the model.
+        
+    Returns
+    -------
+    svg_url_design_matrices: String
+        SVG Image URL for the plotted design matrices,
+    """
+    html_design_matrices = []
+    dmtx_template_path = os.path.join(html_template_root_path,
+                                      'design_matrix_template.html'
+                                      )
+    with open(dmtx_template_path) as html_template_obj:
+        dmtx_template_text = html_template_obj.read()
+    
+    for dmtx_count, design_matrix in enumerate(design_matrices, start=1):
```

Comment:
do we have an example with several design matrices?

Reply:
fiac_flm .

Reply:
thanks!